lib/eib.py

```diff
@@ -242,25 +242,23 @@ def _merge_option(self, section_pattern, option):
                                  opt)
                     yield (section, opt)
             else:
-                add_vals = Counter()
+                values = Counter()
                 for opt in add_opts:
                     logger.debug('Adding %s %s values from %s', section,
                                  option, opt)
-                    add_vals.update(sect[opt].split())
+                    values.update(sect[opt].split())
                     yield (section, opt)
 
-                del_vals = Counter()
                 for opt in del_opts:
                     logger.debug('Removing %s %s values from %s', section,
                                  option, opt)
-                    del_vals.update(sect[opt].split())
+                    values.subtract(sect[opt].split())
                     yield (section, opt)
 
-                # Set the option to the difference of the counters.
+                # Set the option to the keys that have positive values.
                 # Merge the values together with newlines like they were
                 # in the original configuration.
-                vals = add_vals - del_vals
-                sect[option] = '\n'.join(sorted(vals.keys()))
+                sect[option] = '\n'.join(k for k, v in values.items() if v > 0)
```

Comment:
I was curious whether `Counter` provides a shorter way to spell “only keys that have positive values” and it does:


```python
In [7]: values = Counter(foo=3, bar=-1)

In [8]: +values
Out[8]: Counter({'foo': 3})

In [9]: '\n'.join(+values)
Out[9]: 'foo'
```

So we could write:

```suggestion
                sect[option] = '\n'.join(+values)
```

But I've never seen this before and I think it's too obscure.